File: src/lib/avatars.ts
export interface DefaultAvatar {
  id: string
  emoji: string
  bg: string
}

// A fixed set of preset avatars for users who'd rather not upload their own
// photo. `id` is what's stored on the user doc (AppUser.avatarId); `bg` is a
// Tailwind class pair so each preset stays visually distinct.
export const DEFAULT_AVATARS: DefaultAvatar[] = [
  { id: 'bat', emoji: '🏏', bg: 'bg-orange-200 dark:bg-orange-900/50' },
  { id: 'trophy', emoji: '🏆', bg: 'bg-yellow-200 dark:bg-yellow-900/50' },
  { id: 'fire', emoji: '🔥', bg: 'bg-red-200 dark:bg-red-900/50' },
  { id: 'star', emoji: '⭐', bg: 'bg-amber-200 dark:bg-amber-900/50' },
  { id: 'lion', emoji: '🦁', bg: 'bg-yellow-100 dark:bg-yellow-900/40' },
  { id: 'tiger', emoji: '🐯', bg: 'bg-orange-100 dark:bg-orange-900/40' },
  { id: 'eagle', emoji: '🦅', bg: 'bg-blue-200 dark:bg-blue-900/40' },
  { id: 'lightning', emoji: '⚡', bg: 'bg-purple-200 dark:bg-purple-900/40' },
  { id: 'target', emoji: '🎯', bg: 'bg-green-200 dark:bg-green-900/40' },
  { id: 'wave', emoji: '🌊', bg: 'bg-cyan-200 dark:bg-cyan-900/40' },
]

export function getDefaultAvatar(id: string | null | undefined): DefaultAvatar | undefined {
  return id ? DEFAULT_AVATARS.find((a) => a.id === id) : undefined
}

File: android/app/src/main/java/com/mplauction/android/data/Avatars.kt
package com.mplauction.android.data

import androidx.compose.ui.graphics.Color

data class DefaultAvatar(val id: String, val emoji: String, val background: Color)

// Mirrors src/lib/avatars.ts's DEFAULT_AVATARS exactly (same ids, since
// they're what's actually stored on AppUser.avatarId/Team.logoId) — the
// preset a user or team picks instead of uploading a photo.
val DEFAULT_AVATARS =
  listOf(
    DefaultAvatar("bat", "🏏", Color(0xFFFED7AA)),
    DefaultAvatar("trophy", "🏆", Color(0xFFFEF08A)),
    DefaultAvatar("fire", "🔥", Color(0xFFFECACA)),
    DefaultAvatar("star", "⭐", Color(0xFFFDE68A)),
    DefaultAvatar("lion", "🦁", Color(0xFFFEF9C3)),
    DefaultAvatar("tiger", "🐯", Color(0xFFFFEDD5)),
    DefaultAvatar("eagle", "🦅", Color(0xFFBFDBFE)),
    DefaultAvatar("lightning", "⚡", Color(0xFFE9D5FF)),
    DefaultAvatar("target", "🎯", Color(0xFFBBF7D0)),
    DefaultAvatar("wave", "🌊", Color(0xFFA5F3FC)),
  )

fun defaultAvatar(id: String?): DefaultAvatar? = id?.let { i -> DEFAULT_AVATARS.find { it.id == i } }
